operators/src/processing/point_in_polygon.rs

```diff
@@ -656,4 +671,51 @@ mod tests {
 
         assert_eq!(result.len(), 0);
     }
+
+    #[tokio::test]
+    async fn it_checks_sref() {
+        let point_collection =
+            MultiPointCollection::from_data(vec![], vec![], Default::default()).unwrap();
+
+        let polygon_collection = MultiPolygonCollection::from_data(
+            vec![MultiPolygon::new(vec![vec![vec![
+                (0.0, 0.0).into(),
+                (10.0, 0.0).into(),
+                (10.0, 10.0).into(),
+                (0.0, 10.0).into(),
+                (0.0, 0.0).into(),
+            ]]])
+            .unwrap()],
+            vec![TimeInterval::default()],
+            Default::default(),
+        )
+        .unwrap();
+
+        let operator = PointInPolygonFilter {
+            params: PointInPolygonFilterParams {},
+            sources: PointInPolygonFilterSource {
+                points: MockFeatureCollectionSource::with_collections_and_sref(
+                    vec![point_collection],
+                    SpatialReference::epsg_4326(),
+                )
+                .boxed(),
+                polygons: MockFeatureCollectionSource::with_collections_and_sref(
+                    vec![polygon_collection],
+                    SpatialReference::from_str("EPSG:3857").unwrap(),
+                )
+                .boxed(),
+            },
+        }
+        .boxed()
+        .initialize(&MockExecutionContext::test_default())
+        .await;
+
+        match operator {
+            Err(Error::InvalidSpatialReference {
+                expected: _,
+                found: _,
+            }) => {}
+            _ => panic!("Expected error"),
+        }
+    }
```

Comment:
It would be more elegant to use `matches` for that: https://doc.rust-lang.org/std/macro.matches.html

Reply:
669369d8deea6f3adaf10ca146c12b968261faff